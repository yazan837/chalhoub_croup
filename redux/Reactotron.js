import Reactotron from 'reactotron-react-native';
import {reactotronRedux} from 'reactotron-redux';

export default Reactotron.configure({name: 'chalhoub'})
  .useReactNative({
    asyncStorage: false,
    networking: {
      ignoreUrls: /symbolicate/,
    },
    editor: false,
    errors: {veto: stackFrame => false},
    overlay: false,
  })
  .use(reactotronRedux())
  .connect();
