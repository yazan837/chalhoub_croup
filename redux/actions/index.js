import profile from './groups/profile';

export default {
  ...profile,
};
