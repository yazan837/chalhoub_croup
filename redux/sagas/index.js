import {fork} from 'redux-saga/effects';

import * as profile from './groups/profile';

const sagas = Object.values({
  ...profile,
});

export default function* root() {
  for (let index = 0; index < sagas.length; index++) {
    yield fork(sagas[index]);
  }
}
