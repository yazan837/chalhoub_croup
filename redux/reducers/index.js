import {combineReducers} from 'redux';

import profile from './groups/profile';

const appReducers = combineReducers({
  profile,
});

export default (state, action) => {
  return appReducers(state, action);
};
