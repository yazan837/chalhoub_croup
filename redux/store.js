import {createStore, applyMiddleware, compose} from 'redux';
import createSagaMiddleware from 'redux-saga';
import reducers from './reducers';
import sagas from './sagas';

import Reactotron from './Reactotron';

const sagaMiddleware = createSagaMiddleware();
export default createStore(
  reducers,
  {},
  compose(applyMiddleware(sagaMiddleware), Reactotron.createEnhancer()),
);
sagaMiddleware.run(sagas);

//adb reverse tcp:9090 tcp:9090
//adb reverse tcp:8081 tcp:8081
