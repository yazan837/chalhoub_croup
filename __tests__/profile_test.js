import React from 'react';
import {create} from 'react-test-renderer';
import {createStore, compose, applyMiddleware} from 'redux';
import {Provider} from 'react-redux';
import reducers from '../redux/reducers';
import createSagaMiddleware from 'redux-saga';
const sagaMiddleware = createSagaMiddleware();
const store = createStore(
  reducers,
  {},
  compose(applyMiddleware(sagaMiddleware)),
);

import Profile from '../src/profile';
jest.useFakeTimers();
const tree = create(
  <Provider store={store}>
    <Profile />
  </Provider>,
);
test('test profile page ui', () => {
  expect(tree).toMatchSnapshot();
});
