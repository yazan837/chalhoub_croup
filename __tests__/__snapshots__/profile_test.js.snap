// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test profile page ui 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "borderRadius": 60,
          "height": 125,
          "width": 125,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      Name :
    </Text>
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      status:
    </Text>
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      species:
    </Text>
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      gender:
    </Text>
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      location:
    </Text>
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      created:
    </Text>
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    />
  </View>
</View>
`;
